include/xtensor/xpad.hpp

```diff
@@ -81,6 +81,25 @@ namespace xt
 
         using size_type = typename std::decay_t<E>::size_type;
         using value_type = typename std::decay_t<E>::value_type;
+        using type = typename std::remove_reference<E&>::type;
+
+        if (mode == pad_mode::constant)
+        {
+            auto new_shape = e.shape();
+            xt::xstrided_slice_vector sv;
+            for (size_type axis = 0; axis < e.shape().size(); ++axis) {
+                size_type nb = static_cast<size_type>(pad_width[axis][0]);
+                size_type ne = static_cast<size_type>(pad_width[axis][1]);
+
+                size_type ns = nb + e.shape(axis) + ne;
+                new_shape[axis] = ns;
+                sv.push_back(xt::range(nb, nb + e.shape(axis)));
+            }
+
+            type conc_out = constant_value * xt::ones<value_type>(new_shape);
```

Comment:
style (and performances):
```cpp
type conc_out(new_shape, constant_value);
```

The reason is `xt::ones` returns an `xbroadcast` object, which does not provide any method for loading SIMD vector, preventing vectorized assignment (unless the compiler is smart enough to detect it but you have no guarantee of that).